programs/util.c

```diff
@@ -90,6 +90,10 @@ int g_utilDisplayLevel;
 int UTIL_requireUserConfirmation(const char* prompt, const char* abortMsg,
                                           const char* acceptableLetters) {
     int ch, result;
+    /* If input is presented via stdin, dont use prompt as it may swallow characters */
```

Comment:
So there's two considerations to this. One is if input to be compressed is being piped in via stdin. In which case we want to avoid consuming from stdin. But we could test that with `!strcmp(inputFilename, stdinmark)` or whatever. Whereas this also catches the scenario where we are running in a non-interactive environment, and stdin is not a tty, even though input is not being presented through stdin. I think that this is the right behavior there as well, since trying to read from stdin will fail or hang. But it's worth calling that out as well.

Reply:
Thats a good point. I think that checking `inputFilename` against `stdinmark` is the better check. People expect `yes | zstd f1 f2 -o f.zst` to work, even though `-f` would be more conventional. Also, if it doesn't work, that makes it much harder to test.

Reply:
> People expect `yes | zstd f1 f2 -o f.zst` to work

It's not clear to me if this is a valid scenario.
Using `gzip` as a reference : 

```
gzip FILE
FILE.gz already exists -- do you wish to overwrite (y or n)? y

yes | gzip FILE
gzip: FILE.gz already exists -- skipping
```

Reply:
I expect tools that accept yes/no input to work with `yes`. I'm surprised when they don't. When I'm using an unfamiliar API, it can be easier to just pipe in `yes` than to try to figure out which option suppresses the warning, if any.

I think it would also be hard to test that the prompt works correctly if the input must be a console, meaning all the existing test cases we have where we `echo y | zstd` wouldn't work.